app/controllers/application_controller.rb

```diff
@@ -316,4 +316,12 @@ def mobile?
     client = DeviceDetector.new(request.user_agent)
     client.device_type != 'desktop'
   end
+
+  def x509_presented?
+    current_sp&.attribute_bundle&.include?('x509_presented')
```

Comment:
`attribute_bundle` is set in the DB ahead of time at a per-SP level, not per-request. I think we should be looking at the current ServiceProviderRequest (`service_provider_request`) instead

Reply:
When coming from the sample SP at `localhost:9292`, when I get to this line of code and check the service_provider_request, it's unavailable `<NullServiceProviderRequest:0x00007fae94bd9d50>`.  Is there a way to set things up locally so the sp request actually works?

Also, assuming it did work, would I do something like `service_provider_request.requested_attributes.include?('x509_presented')`?

Reply:
Looks like `ApplicationController#service_provider_request` uses `params[:request_id]` which is only available in a few places (because it checks that page's request params)

Looks like in other places we do:

```
ServiceProviderRequestProxy.from_uuid(sp_session[:request_id])
```

which pull the request_id from the session, which is longer lived than an individual request --- try that?

Reply:
But yes `requested_attributes.include?('x509_presented')` is what I'd go with